pymatgen/io/openmm/utils.py

```diff
@@ -360,111 +360,182 @@ def parameterize_system(
-        forcefield_omm = omm_ForceField()
+                mol_to_load = openff_mol
+            # Assign ff to each molecule
+            if smile in force_field.keys():
+                ff_name = force_field[smile]
+                if ff_name.lower() in all_small_ffs or ff_name.lower() in basic_small_ffs:
+                    small_molecules[mol_to_load] = ff_name.lower()
+                else:
+                    large_or_water[mol_to_load] = ff_name.lower()
+            else:
+                small_molecules[mol_to_load] = "sage"
+
+        # Determine category of forcefield for deciding which water model
+        large_ff_category = ""
+        for ff in large_or_water.values():
+            temp_ff_string = ""
+            if "amber" in ff.lower():
+                temp_ff_string = "amber"
+            elif "charmm" in ff.lower():
+                temp_ff_string = "charmm"
+            elif "amoeba" in ff.lower():
+                temp_ff_string = "amoeba"
+            else:
+                continue
+            if large_ff_category == "":
+                large_ff_category = temp_ff_string
+            else:
+                if large_ff_category != temp_ff_string:
+                    warnings.warn(f"Did you mean to mix {temp_ff_string} and " f"{large_ff_category} force fields?")
+
+        ff_to_load = ""
         for ff in large_or_water.values():
-            forcefield_omm.loadFile(ff)
-        for mol, ff in small_molecules.items():
-            if "gaff" in ff:
+            if ff in basic_water_ffs:
+                # Ensure the water model matches the large molecule model
+                if large_ff_category:
+                    if large_ff_category == "amber":
+                        if ff == "spce":
+                            ff_to_load = "amber14/spce.xml"
+                        if ff == "tip3p":
+                            ff_to_load = "amber14/tip3p.xml"
+                        if ff == "tip4p":
+                            ff_to_load = "amber14/tip4pew.xml"
+                    elif large_ff_category == "charmm":
+                        if ff == "spce":
+                            ff_to_load = "charmm36/spce.xml"
+                        if ff == "tip3p":
+                            ff_to_load = "charmm36/water.xml"
+                        if ff == "tip4p":
+                            ff_to_load = "charmm36/tip4pew.xml"
+                    else:
+                        warnings.warn(f"Did you mean to use {ff} with the " f"{large_ff_category} force field?")
+                # If there isn't a large molecule forcefield required,
+                # assume amber14
+                else:
+                    if ff == "spce":
+                        ff_to_load = "amber14/spce.xml"
+                    if ff == "tip3p":
+                        ff_to_load = "amber14/tip3p.xml"
+                    if ff == "tip4p":
+                        ff_to_load = "amber14/tip4pew.xml"
```

Comment:
If there isn't a way to simplify this I'd suggest wrapping it in a function called `assign_water_forcefield` and sticking it in `utils.py`. I think we should try to keep the code for `parameterize_system` a bit more abstract.

Reply:
Added a lookup dictionary